docs/developer/extending/apps/architecture/communication-between-app-and-saleor.mdx

```diff
@@ -32,12 +32,24 @@ sequenceDiagram
 
 To accept webhooks, your app must include public endpoints listening for HTTP POST requests. You have to also ensure the app [subscribes to the right events](developer/extending/webhooks/asynchronous-events.mdx).
 
-Event delivery will retry if:
+#### Time limits
 
-- the endpoint does not respond before timeout
-- the endpoint will respond with a non-200 status code
+Webhook endpoints must respond within 20 seconds, including the network latency.
 
-If your application performs long-running operations, you should consider using the task queue.
+If your application performs long-running operations, consider using a [task queue](#task-queue) instead.
+
+#### Fault tolerance
+
+Event delivery will be retried up to 5 times with exponential backoff if:
```

Comment:
non blocking, but we can mention what are actual intervals and how to control them (can it be set via env?)

Reply:
There is no way to retry synchronous webhooks as they are synchronous.

Reply:
@lkostrowski I don't want to document that. Actual delivery time will depend on traffic (queue lengths), and we don't want to offer a way to increase backoff as delayed tasks use a lot of memory due to how Celery works.